File: C++ language/Bill.cpp
#include <iostream>
#include <string>

using namespace std;

// Interface definition (in C++, typically handled via abstract classes with pure virtual functions)
class Votable {
public:
    virtual void vote() = 0; // Pure virtual function
};

// Bill class implementation
class Bill : public Votable {
private:
    string title;
    string status;
    int votes;
    int billId;

public:
    Bill(string title, int billId, string status) : title(title), billId(billId), status(status), votes(0) {}

    string getTitle() {
        return title;
    }

    void setTitle(string title) {
        this->title = title;
    }

    string getStatus() {
        return status;
    }

    void setStatus(string status) {
        this->status = status;
    }

    int getVotes() {
        return votes;
    }

    void setVotes(int votes) {
        this->votes = votes;
    }

    int getBillID() {
        return billId;
    }

    void setBillID(int billId) {
        this->billId = billId;
    }

    // Override vote function
    void vote() override {
        cout << "Voting on bill '" << title << "'" << endl;
        votes++;
    }
};

File: C# language/Bill.cs
public class Bill : IVotable
{
    private string title;
    private string status;
    private int votes;
    private int billId;

    public Bill(string title, int billId, string status)
    {
        this.title = title;
        this.status = status;
        this.billId = billId;
        this.votes = 0;
    }

    public string Title
    {
        get { return title; }
        set { title = value; }
    }

    public string Status
    {
        get { return status; }
        set { status = value; }
    }

    public int Votes
    {
        get { return votes; }
        set { votes = value; }
    }

    public int BillId
    {
        get { return billId; }
        set { billId = value; }
    }

    public void Vote()
    {
        Console.WriteLine("Voting on bill '" + title + "'");
        votes++;
    }
}
